jenkins/catapult/jenkins/catapult-client-build-catapult-project.groovy

```diff
@@ -261,7 +261,7 @@ pipeline {
 								docker.image(baseImageNames[0]).inside("--volume=${pwd()}/catapult-src:/catapult-src") {
 									sh '''
 										cd /catapult-src
-										lcov --directory client/catapult/_build --capture --output-file coverage_all.info
+										lcov --directory client/catapult/_build --capture --output-file coverage_all.info --ignore-errors mismatch
```

Comment:
why did you need to add `--ignore-errors mismatch` 🤔

Reply:
In lcov 2 error handling was added.  Now, there are some errors in the tests.
But also errors in the ``/usr`` and boost header files. 

```
b) Error handling

     A generic - but very simple - error handler has been added to the
     lcov tool suite.  The error handler is used to report exceptions,
     and provides a mechanism for the user to ignore the particular
     message if desired.
     See the genhtml/lcov/geninfo man pages for details.

     Note that some errors are unrecoverable - and cannot be suppressed or
     ignored.

     Related options:  --ignore-error, --keep-going
```

```
geninfo: WARNING: ('mismatch') mismatched end line for _ZN8catapult8builders64MosaicAddressRestrictionBuilderTests_CanSetMosaicId_Regular_TestD0Ev at /catapult-src/client/catapult/sdk/tests/builders/MosaicAddressRestrictionBuilderTests.cpp:99: 108 -> 99
	(use "geninfo --ignore-errors mismatch,mismatch ..." to suppress this warning)
geninfo: WARNING: ('mismatch') mismatched end line for _ZN8catapult8builders64MosaicAddressRestrictionBuilderTests_CanSetMosaicId_Regular_Test8TestBodyEv at /catapult-src/client/catapult/sdk/tests/builders/MosaicAddressRestrictionBuilderTests.cpp:99: 108 -> 99
	(use "geninfo --ignore-errors mismatch,mismatch ..." to suppress this warning)
```

```
Processing client/catapult/_build/tests/int/node/basic/CMakeFiles/tests.catapult.int.node.dir/LocalNodeRequestTests.cpp.gcda
geninfo: WARNING: /usr/include/c++/13/bits/std_function.h:86: unexecuted block on non-branch line with non-zero hit count.  Use "geninfo --rc geninfo_unexecuted_blocks=1 to set count to zero.
	(use "geninfo --ignore-errors gcov,gcov ..." to suppress this warning)
Processing client/catapult/_build/tests/int/node/basic/CMakeFiles/tests.catapult.int.node.dir/LocalNodeApiBasicTests.cpp.gcda
geninfo: WARNING: /mybuild/include/boost/smart_ptr/detail/shared_count.hpp:448: unexecuted block on non-branch line with non-zero hit count.  Use "geninfo --rc geninfo_unexecuted_blocks=1 to set count to zero.
```